### PR TITLE
fix: decoding of update_schedule_enclave

### LINKED ISSUE #1828: Extrinsic `update_scheduled_enclave` doesn't work
### Context

It occurred when I tried to set the scheduled enclave in parachain on staging. When parsing and executing the extrinsic in parentchain block it says:

```
[2023-06-27T07:07:54Z WARN  itc_parentchain_indirect_calls_executor::filter_calls] Could not decode. Error { cause: Some(Error { cause: None, desc: "Not enough data to fill buffer" }), desc: "Could not decode `UpdateScheduledEnclaveArgs::mrenclave`" }
```

Can you please take a look at it or even test it locally?

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by felixfaisal**
Resolves #1828 
I tested it locally

### CODE REVIEW

File: tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/scheduled_enclave.rs [hunk 1]
```diff
@@ -25,15 +25,15 @@ use log::debug;
 
 #[derive(Debug, Clone, Encode, Decode, Eq, PartialEq)]
 pub struct UpdateScheduledEnclaveArgs {
-	sbn: SidechainBlockNumber,
+	sbn: codec::Compat<SidechainBlockNumber>,
```
Comment (kziemianek) [inline tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/scheduled_enclave.rs]: There is same type used in `RemoveScheduledEnclaveArgs`, doesn't it need to be fixed too ?

Comment (felixfaisal) [reply tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/scheduled_enclave.rs]: Yeah I thought the same thing but the compact is being used only for `update_scheduled_enclave` 
```rust
		pub fn update_scheduled_enclave(
			origin: OriginFor<T>,
			#[pallet::compact] sidechain_block_number: SidechainBlockNumber,
			mr_enclave: MrEnclave,
		) -
```
and not in `remove_scheduled_enclave` [code](https://github.com/litentry/litentry-parachain/blob/9c817a781d5c2d6659f978933958a79531a84fb2/pallets/teerex/src/lib.rs#L504C1-L507C36)
```rust
		pub fn remove_scheduled_enclave(
			origin: OriginFor<T>,
			sidechain_block_number: SidechainBlockNumber,
		) -> DispatchResultWithPostInfo {
```
I don't know why we don't include `compact` in this extrinsic as well. 
CC: @songzhouran @Kailai-Wang

Comment (Kailai-Wang) [reply tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/scheduled_enclave.rs]: Yes we should apply `compact` too 😂

Comment (felixfaisal) [reply tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/scheduled_enclave.rs]: Done, Waiting for CI to pass.

**Review by Kailai-Wang — APPROVED** [review]

**Pull request merged**